promql/quantile.go

```diff
@@ -158,9 +158,18 @@ func histogramQuantile(q float64, h *histogram.FloatHistogram) float64 {
 	var (
 		bucket histogram.Bucket[float64]
 		count  float64
-		it     = h.AllBucketIterator()
-		rank   = q * h.Count
+		it     histogram.BucketIterator[float64]
+		rank   float64
 	)
+
+	if math.IsNaN(h.Sum) || q < 0.5 {
```

Comment:
Maybe add a comment here to explain the special NaN case because the reason is rather obscure.